packages/std/src/math/uint128.rs

```diff
@@ -286,14 +286,14 @@ impl ops::Mul<Uint128> for Uint128 {
         )
     }
 }
+forward_ref_binop!(impl Mul, mul for Uint128, Uint128);
 
-impl<'a> ops::Mul<&'a Uint128> for Uint128 {
-    type Output = Self;
-
-    fn mul(self, rhs: &'a Uint128) -> Self::Output {
-        self.mul(*rhs)
+impl MulAssign<Uint128> for Uint128 {
+    fn mul_assign(&mut self, rhs: Self) {
+        *self = *self * rhs;
```

Comment:
```suggestion
        *self *= rhs;
```
?
And below, if there's nothing wrong with this.

Reply:
with this change you'd implement `mul_assign` with `mul_assign`